VideoLocker/src/main/java/org/edx/mobile/task/Task.java

```diff
@@ -49,6 +45,7 @@ public Task(Context context) {
 
     public void setProgressDialog(ProgressBar progressBar) {
         this.progressBar = progressBar;
+        this.progressCallback = null;
```

Comment:
This doesn't _need_ to be automatically exclusive, but I suppose it can be set up that way. I certainly don't see any use case for using both a callback and a directly controller `ProgressBar` at the same time. However, to be consistent then, you should make them mutually exclusive (i.e. `setProgressCallback()` should also set `progressBar` to `null`).